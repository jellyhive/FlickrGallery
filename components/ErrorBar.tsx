import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

type ErrorBarProps = {
    visible: boolean
    text?: string
}

const ErrorBar: React.FC<ErrorBarProps> = ({visible, text}) => {

    return (
        <>
        { visible && <View style={style.container}>
                        <Text style={style.text}>{text}</Text>
                    </View> }
        </>
    )
}
export default ErrorBar

const style = StyleSheet.create({
    container: {
      width: '100%',
      backgroundColor: '#ff0000'
    },
    text: {
        width: '100%',
        paddingVertical: 10,
        paddingHorizontal: 20,
        paddingBottom: 10,
        color: '#ffffff'
      }
  })