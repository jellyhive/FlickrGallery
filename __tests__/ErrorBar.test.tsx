import React from 'react'
import ErrorBar from '../components/ErrorBar'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
    let tree: any;
    tree = renderer.create(<ErrorBar visible text="Error message" />).toJSON()
    expect(tree).toMatchSnapshot()
})