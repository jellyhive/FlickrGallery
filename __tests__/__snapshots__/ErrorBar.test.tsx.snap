// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ff0000",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "paddingBottom": 10,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    Error message
  </Text>
</View>
`;
